### PR TITLE
Replaces encoding/json with easyjson

### PR DESCRIPTION
**Pull request by masseelch**
This PR replaces the `encoding/json` package and removes the dependency on [sheriff](github.com/liip/sheriff) and eliminates two major reflection-usages in this extension. 

`elk` now creates response structs for every endpoint as requested and defined by the `serialization groups` on the schema and executes the [easyjson](https://github.com/mailru/easyjson) generator on those structs.

### CODE REVIEW

File: easyjson.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+func GenerateEasyJSON(next gen.Generator) gen.Generator {
+	return gen.GenerateFunc(func(g *gen.Graph) error {
+		// Let ent create all the files.
+		if err := next.Generate(g); err != nil {
+			return err
+		}
+
+		// We want to render every response struct created with easyjson.
+		ns := make([]string, 0)
```
Comment (a8m) [inline easyjson.go:17]:
```suggestion
		var ns []string
```

Or use make with `len(g.Nodes) * 4`

File: edges.go [hunk 2]
```diff
@@ -67,20 +59,17 @@ func edgesToLoad(n *gen.Type, action string) (*eagerLoadEdges, error) {
...
 }
 
-// edgesToLoadHelper returns the query to use to eager load all edges as required by the annotations defined
-// on the given node.
-func edgesToLoadHelper(n *gen.Type, visited map[string]uint, groupsToLoad []string) (*eagerLoadEdges, error) {
+// edgesToLoadHelper recursively collects the edges to load on this type for requested groups on the given action.
+func edgesToLoadHelper(n *gen.Type, visited map[string]uint, groupsToLoad []string) (EdgesToLoad, error) {
 	// What edges to load on this type.
-	edges := make([]eagerLoadEdge, 0)
+	edges := make(EdgesToLoad, 0)
```
Comment (a8m) [inline edges.go:72]:
```suggestion
	var edges []EdgesToLoad
```

**Review by a8m — APPROVED** [review]

File: easyjson.go (hunk 1 above)
Comment (masseelch) [reply easyjson.go:17]: Thanks!

File: edges.go (hunk 2 above)
Comment (masseelch) [reply edges.go:72]: :bow:

**Pull request merged**